src/main/java/com/zendesk/maxwell/producer/partitioners/HashStringTable.java

```diff
@@ -9,6 +9,12 @@
  */
 public class HashStringTable implements HashStringProvider {
 	public String getHashString(RowMap r, List<String> partitionColumns, String partitionKeyFallback) {
-		return r.getTable();
+		String t = r.getTable();
+
+		// support table-to-database fallback for DDL that has no table associated
+		if ( t == null && partitionKeyFallback.equals("database") )
```

Comment:
maybe `"database".equals(partitionKeyFallback)` in case `partitionKeyFallback` is `null`